modules/syslogformat/syslog-parser.c

```diff
@@ -36,8 +44,16 @@ syslog_parser_process(LogParser *s, LogMessage **pmsg, const LogPathOptions *pat
             evt_tag_str ("input", input),
             evt_tag_printf("msg", "%p", *pmsg));
 
-  msg_format_parse(&self->parse_options, msg, (guchar *) input, input_len);
-  return TRUE;
+  if (self->drop_invalid)
```

Comment:
Wouldn't improve things a `G_UNLIKELY` here? Just like we do in `LogParser` with template option

Reply:
I can add this.

Reply:
Hmm.. I have second thoughts on this. A config may easily use drop_invalid mode exclusively, in which case G_UNLIKELY is not true.

In the case of the "template" option, that branch is slow anyway, so the G_UNLIKELY is the least of the concerns anyway.

So I would just leave it to the compiler.

Reply:
I see, so this wouldn't make too much sense then.